misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/RangeFormatterTest.java

```diff
@@ -38,46 +44,32 @@
 /**
  * The abstract class that is extended by all range formatting test classes.
  *
- * @since 1.2.10
+ * @since 2.0.0
  */
-public abstract class RangeFormatter {
+public abstract class RangeFormatterTest {
     private final Path resourceDirectory = Paths.get("src").resolve("test").resolve("resources").toAbsolutePath();
-    private Path buildDirectory = Paths.get("build").toAbsolutePath().normalize();
     private static final String ASSERT_DIR = "assert";
     private static final String SOURCE_DIR = "source";
 
+    private static final Gson gson = new Gson();
+
     @Test(dataProvider = "test-file-provider")
-    public void test(String source, int[][] positions)
+    public void test(String sourcePath, String source, JsonArray positions)
```

Comment:
What is the difference between the `source` and the `sourcePath`?

Reply:
The method signature was referred from the currently implemented `FormatterTest` class as following: 

https://github.com/ballerina-platform/ballerina-lang/blob/155a4957fa9ec4415cde7c44ca5c3098788e296a/misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/FormatterTest.java#L56 

- The `source` is the file name. 
  - e.g.: _"checking_action_01.bal"_

- The `sourcePath` is the relative directory where the testcase related files are in (relative to the `resource` directory). 
  - e.g.: "actions/checking"

Reply:
In the newly suggested approach, this test function signature was changed to `test(Path sourceFilePath, Path assertFilePath, ArrayList<LineRange> lineRanges)` whereas it has been implemented here.